sources/packages/backend/apps/api/src/testHelpers/auth/institution-user-helpers.ts

```diff
@@ -0,0 +1,26 @@
+import { TestingModule } from "@nestjs/testing";
+import { getProviderInstanceForModule } from "@sims/test-utils";
+import { AuthModule } from "../../auth/auth.module";
+import { InstitutionUserAuthService } from "../../services";
+import { InstitutionUserAuthorizations } from "../../services/institution-user-auth/institution-user-auth.models";
+
+/**
+ * Mocks institution user authorizations that will be required to authorize the token.
+ * @param testingModule nest testing module.
+ * @param institutionUserAuthorizations authorizations.
+ */
+export async function mockInstitutionUserAuthorization(
+  testingModule: TestingModule,
+  institutionUserAuthorizations: InstitutionUserAuthorizations,
+): Promise<void> {
+  const institutionUserAuthService = await getProviderInstanceForModule(
+    testingModule,
+    AuthModule,
+    InstitutionUserAuthService,
+  );
+  jest
+    .spyOn(institutionUserAuthService, "getAuthorizationsByUserName")
+    .mockImplementation(() => {
+      return Promise.resolve(institutionUserAuthorizations);
+    });
```

Comment:
Is there any particular reason not to follow the same done for the student mock?
```ts
institutionUserAuthService.getAuthorizationsByUserName = jest
    .fn()
    .mockResolvedValue(institutionUserAuthorizations);
```

Reply:
Yes. When I was doing the implementation , I was trying to ensure how to restore the mock, in case if any test written after the test which mocks the authorizations and the new test require the original method implementation and not the mocked one. 

I found that it can be done by `jest.restoreAllMocks()` only when mock is created with `jest.spyOn`

![image](https://github.com/user-attachments/assets/91839ec5-e6c9-4df0-929e-b0caf19d0cc4)